examples/asyncimage/asyncimage.go

```diff
@@ -0,0 +1,194 @@
+	if dynamicImage.GetState() == g.SsLoading {
+		return TextHeader("Image Is Currently loading...", color.RGBA{0x80, 0x80, 0xFF, 255})
+	}
+	return CanLoadHeader()
+}
+
+func FooterWidget(label string) g.Widget {
+	return g.Row(TextHeader(label, color.RGBA{0xFF, 0xFF, 0xFF, 255}))
+}
+
+func ShouldReturnImage() g.Widget {
+	if dynamicImage.GetState() != g.SsSuccess {
+		return fallback.ToImageWidget().Size(-1, -1)
+	}
+	return g.Custom(func() { dynamicImage.ToImageWidget().Scale(imageScaleX, imageScaleY).Build() })
+}
+
+func ShouldReturnPanel() g.Widget {
+
+	return g.Custom(func() {
+		imgui.SeparatorText("Image Scale")
+		if imgui.Button("Reset##Scaling") {
+			imageScaleX = 1.0
+			imageScaleY = 1.0
+		}
+		imgui.SameLine()
+		imgui.Checkbox("Linked##Scaling", &linkedScale)
+		if linkedScale {
+			imgui.SliderFloat("scale XY##Scaling", &imageScaleX, 0.1, 4.0)
+			imageScaleY = imageScaleX
+		} else {
+			imgui.SliderFloat("scale X##Scaling", &imageScaleX, 0.1, 4.0)
+			imgui.SliderFloat("scale Y##Scaling", &imageScaleY, 0.1, 4.0)
+		}
+		imgui.SeparatorText("FileSystem URLS")
+		if dynamicImage.GetState() == g.SsLoading {
+			imgui.Text("Unavailable while loading image...")
+		} else {
+			imgui.Text("Loading URLS Works with file:/// scheme too.")
+			imgui.Text("By default, root is executable working dir")
+			imgui.Text("-> Try loading this in the url bar:")
+			imgui.Text("file:///files/sonic.png ->")
+			if imgui.Button("or CLICK HERE") {
+				inputRootfs = "."
+				dynamicImage.SetFSRoot(inputRootfs)
+				dynamicImageUrl = "file:///files/sonic.png"
+				err := dynamicImage.ResetState()
+				if err == nil {
+					dynamicImage.SetSurfaceFromURL(dynamicImageUrl, time.Second*5, false)
+				}
+				linkedScale = true
+				imageScaleX = 0.356
+				imageScaleY = 0.356
+			}
+			imgui.Separator()
+			imgui.Text("Set rootFS to / for full filesystem access")
+			rootfs := dynamicImage.GetFSRoot()
+			imgui.Text(fmt.Sprintf("Current ROOTFS: %s", rootfs))
+			g.InputText(&inputRootfs).Build()
+			imgui.SameLine()
+			if imgui.Button("SET rootfs") {
+				dynamicImage.SetFSRoot(inputRootfs)
+			}
+		}
+	})
+
+}
+
+func loop() {
+
+	if !showWindow {
+		wnd.SetShouldClose(true)
+	}
+
+	g.PushColorWindowBg(color.RGBA{30, 30, 30, 255})
+	g.Window("Async Images").IsOpen(&showWindow).Pos(10, 30).Size(1280, 720).Flags(g.WindowFlagsNoResize).Layout(
+		HeaderWidget(),
+		g.Separator(),
+		g.Row(
+			g.Child().Size(400, 625).Layout(
+				ShouldReturnPanel(),
+			),
+			g.Child().Flags(g.WindowFlagsHorizontalScrollbar).Size(-1, 625).Layout(
+				ShouldReturnImage(),
+			)),
+		g.Separator(),
+		FooterWidget(footerLabel),
+	)
+	g.PopStyleColor()
+}
+
+func noOSDecoratedWindowsConfig() g.MasterWindowFlags {
+	imgui.CreateContext()
+	implot.PlotCreateContext()
+	imnodes.ImNodesCreateContext()
+	io := imgui.CurrentIO()
+	io.SetConfigViewportsNoAutoMerge(true)
+	io.SetConfigViewportsNoDefaultParent(true)
+	io.SetConfigWindowsMoveFromTitleBarOnly(true)
```

Comment:
generally this is not a valid use of CreateContext as it is also created by glfw backend.
Try to call this function after giu.CreateMasterWindow
and do not initialize plot/imnodes - giu should do that.

Reply:
Got rid of impot/imnodes createcontext
But after some try trying to SetConfigViewports after the window creation lead to the window initially merged with main viewport, defeating purpose. That's hacky there but I got no clean solution in current workflow :/